engine/glfw/lib/cocoa/cocoa_window.m

```diff
@@ -626,19 +626,16 @@ int  _glfwPlatformOpenWindow( int width, int height,
     if (_glfwWin.clientAPI == GLFW_OPENGL_API)
     {
         unsigned int attribute_count = 0;
-    #define ADD_ATTR(x) attributes[attribute_count++] = x
-    #define ADD_ATTR2(x, y) (void)({ ADD_ATTR(x); ADD_ATTR(y); })
     #define MAX_ATTRS 24 // urgh
         NSOpenGLPixelFormatAttribute attributes[MAX_ATTRS];
+    #define ADD_ATTR(x) assert(attribute_count < MAX_ATTRS); attributes[attribute_count++] = x
+    #define ADD_ATTR2(x, y) (void)({ ADD_ATTR(x); ADD_ATTR(y); })
```

Comment:
Added an assert to a void a potential buffer overrun.
(it wasn't an issue, I just felt it a bit cleaner)